rust/xaynet-server/src/storage/api.rs

```diff
@@ -0,0 +1,240 @@
+//! Storage API.
```

Comment:
all the getters of the traits in this module take the implementer by `&mut self`. is that required by the types that usually would implement the traits? if so we should probably document this somewhere, otherwise i would expect the getters to just take `&self`.

Reply:
I've made them `mut` just in case we want to use `tower` later, as the `tower` `call` method requires `mut`.